modules/benchmark/R/metric.residual.plot.R

```diff
@@ -8,21 +8,24 @@
 metric.residual.plot <- function(dat, var){
   
   require(ggplot2)
+  # 
+  # ind <- intersect(which(!is.na(dat$obvs)),  which(!is.na(dat$model)))
+  # 
+  # dat <- dat[ind,]
   
-  ind <- intersect(which(!is.na(dat$obvs)),  which(!is.na(dat$model)))
-  
-  dat <- dat[ind,]
-  dat$time <- lubridate::year(as.Date(as.character(dat$time), format="%Y",))
-  dat$diff <- dat$model - dat$obvs
+  dat$time <- lubridate::year(as.Date(as.character(dat$time), format="%Y"))
+  dat$diff <- abs(dat$model - dat$obvs)
+  dat$zeros <- rep(0, length(time))
   
-  ggplot(data = dat) + 
-    geom_path(aes(x=time,y=rep(0, length(time))), colour = "#666666", size=2, linetype = 2, lineend = "round") +
-    geom_point(aes(x=time,y=diff), size=4,  colour = "#619CFF") + labs(title=var, x= "years", y="model - observation")
+  p <- ggplot(data = dat, aes(x=time)) + 
+    geom_path(aes(y=zeros), colour = "#666666", size=2, linetype = 2, lineend = "round") +
+    geom_point(aes(y=diff), size=4,  colour = "#619CFF") + labs(title=var, x= "years", y="abs(model - observation)")
   
+  plot(p)
   # ind <- intersect(which(!is.na(dat$obvs)),  which(!is.na(dat$model)))
   # plot(dat$model[ind]-dat$obvs[ind], ylim = c(-max(dat$model[ind]-dat$obvs[ind]),max(dat$model[ind]-dat$obvs[ind])))
   # abline(h=0)
-  
+
   return(NA)
```

Comment:
I like the idea of working towards 2, but we don't want to restrict metrics to just returning either scalars or ggplot objects. I really liked the idea that we could use the file variant to also store metrics that returned vectors, matrices, etc, which we'd loose if we too strictly assumed the metric was returning a plot.  Plus, old folk like me still make plots using base R